frontend/actions/src/main/kotlin/com/virtuslab/gitmachete/frontend/actions/dialogs/SlideInDialog.kt

```diff
@@ -172,12 +175,31 @@ class SlideInDialog(
 
       add(branchField, ComponentConstraint().minWidth("${JBUI.scale(300)}px").growX().wrap())
 
+      add(
+        JLabel(
+          getString(
+            "action.GitMachete.BaseSlideInBelowAction.dialog.slide-in.label.custom-annotation",
+          ),
+        ),
+        ComponentConstraint().gapAfter("0").minWidth("${JBUI.scale(100)}px"),
+      )
+
+      add(customAnnotationField, ComponentConstraint().minWidth("${JBUI.scale(300)}px").growX().wrap())
+
       add(reattachCheckbox)
     }
   }
 
-  private fun createBranchField() =
-    ComboBoxWithAutoCompletion(MutableCollectionComboBoxModel(mutableListOf<String>()), project)
+  private fun createBranchField(): ComboBoxWithAutoCompletion<String> {
+    val model = object : MutableCollectionComboBoxModel<String>(mutableListOf<String>()) {
+      override fun setSelectedItem(item: Any?) {
```

Comment:
a note here: I am "tracking" the selection change in the combobox model and and change the value of the custom annotation field to the value coming from the selected branch.

e.g. 
1. having layout:
```
develop 
  foo #PR-2137
master
```
2. slide in below `master`
3. select `foo` from the combo box
 
result:
`#PR-2137` appears as the custom annotation value ! 

one can edit it later ofc but I assumed it is better to keep the value by default

Reply:
Actually, pls mention that in a code comment (although it can be deduced pretty easily from code, methinks)

Reply:
Whoops it doesn't seem to work for me hmmm

<img width="954" alt="image" src="https://user-images.githubusercontent.com/3383210/216840427-0ac07f8e-83af-4831-b091-2cec8cec9241.png">

I'd expect the existing annotation for `build-chain` to show up in text field... but it doesn't happen ☹️

Reply:
did you pick the branch from the list or write it?

Reply:
Huh probably typed in like 2 first letters and then selected from the dropdown... what is the expected behavior in either case?

Reply:
1. expand the dropdown
2. click branch `foo` 

expected: foo's annotation appears 

1. write `foo` in the branch field

expected: foo's annotation does not appear

Reply:
pls provide a specific steps to reproduce

Reply:
https://user-images.githubusercontent.com/3383210/217201857-4e7439fc-a99e-4246-a4fd-cdcbe64a64bd.mov

So yeah, when I select without typing, then the annotation shows up... but when I type anything and then select, then no annotation shows up ☹️

Reply:
pls try it now

Reply:
Seems to work now indeed 🎉